code/controllers/subsystem/input.dm

```diff
@@ -117,6 +117,5 @@ SUBSYSTEM_DEF(input)
 	var/list/clients = GLOB.clients // Let's sing the list cache song
 	if(listclearnulls(clients)) // clear nulls before we run keyloop
 		log_world("Found a null in clients list!")
-	for(var/i in 1 to clients.len)
-		var/client/C = clients[i]
+	for(var/client/C in clients)
```

Comment:
There's another loop in here that uses the `i in 1 to clients.len`

Reply:
I don't understand what you mean

Reply:
Actually, you might be onto something here with the typeless loops. The idea was that using typed loops would skip over null entries, but this would be vulnerable to an excessive quantity of nulled clients, as it wouldn't clean up the list if it got clogged with a bunch of entries. Additionally, typeless loops have a minor performance benefit, although for something that runs on the order of 100 entries 10 times a second, it's not likely to be anything perceptible.

Reply:
What method would be preferred in this case? typeless or typed?

Reply:
Typeless and purging nulls beforehand would be the way to go about it

Reply:
will revert then